# Sample exam 3

Just generic python setup first, including some helper functions to help with degree-radians conversion, and solving quadratic equations

In [2]:
import numpy as np
from numpy import sin, cos, tan, arctan2, sqrt

In [3]:
def rad(th):
    return th * np.pi/180

def deg(th):
    return th * 180/np.pi

def quadratic(a, b, c):
    return ((-b + sqrt(b**2 - 4*a*c)) / (2*a),
            (-b - sqrt(b**2 - 4*a*c)) / (2*a))

## Problem 1

### Part (a)

Momentum is (approximately) conserved in this situation, as it is basically a (completely inelastic) collision. Catching the snow ball happens quickly, and the external forces (gravity, normal force, friction) can be neglected over that short time, in as far as they don't cancel out already, anyway (normal force and gravity on Cartman cancel, anyway)

### Part (b)

So we'll use $\vec p_i = \vec p_f$. I'm using the subscripts $C$ and $S$ for Cartman and the snowball, respectively
$$
m_C \vec v_{Ci} + m_S \vec v_{Si} = (m_C + m_S) \vec v_f \Longrightarrow \vec v_f = \frac{m_C \vec v_{Ci} + m_S \vec v_{Si}}{m_C + m_S}
$$
We have to write our vectors in components, and plug them in.
\begin{align}
\vec v_{Ci} &= v_C (-\cos 45^\circ \hat i + \sin 45^\circ \hat j)\\
\vec v_{Si} &= v_S (\cos 30^\circ \hat i + \sin 30^\circ \hat j)\\
\end{align}

In [15]:
m_C = 100
v_C = 5
m_S = 20
v_S = 10
p_Ci = m_C * v_C * np.array([-cos(rad(45)), sin(rad(45))])
p_Si = m_S * v_S * np.array([cos(rad(30)), sin(rad(30))])
p_i = p_Ci + p_Si
p_f = p_i
v_f = p_f / (m_C + m_S)
print("v_f in components: {:f} mph, {:f} mph".format(v_f[0], v_f[1]))
print("speed v_f {:f} mph angle {:f}".format((v_f[0]**2+v_f[1]**2)**.5, deg(np.arctan2(v_f[1], v_f[0]))))

v_f in components: -1.502903 mph, 3.779612 mph
speed v_f 4.067454 angle 111.684457


So 4.1 mph $112^\circ$ counterclockwise from the x (east) axis, that is, $22^\circ$ W of N.